modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoDataStore.java

```diff
@@ -211,13 +224,18 @@ private MongoSchemaStore createSchemaStore(String schemaStoreURI) {
     final FilterCapabilities createFilterCapabilties() {
         FilterCapabilities capabilities = new FilterCapabilities();
 
-        /* disable FilterCapabilities.LOGICAL_OPENGIS since it contains
-            Or.class (in addtions to And.class and Not.class.  MongodB 2.4
-            doesn't supprt '$or' with spatial operations.
-        */
-        //        capabilities.addAll(FilterCapabilities.LOGICAL_OPENGIS);
-        capabilities.addType(And.class);
-        capabilities.addType(Not.class);
+        if (deactivateOrNativeFilter) {
+            /*
+             * disable FilterCapabilities.LOGICAL_OPENGIS since it contains Or.class (in
+             * additions to And.class and Not.class. MongodB 2.4 doesn't support '$or' with
+             * spatial operations.
+             */
+            capabilities.addType(And.class);
+            capabilities.addType(Not.class);
```

Comment:
So the goal is to remove the ``Or.class``, I would have extended ``FilterCapabilities`` with a remove type function, although not sure how that would play out with the ``FUNCTIONS``. 

Anyway, ok by me, no need to change, this is just a note.